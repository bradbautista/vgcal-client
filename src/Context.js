import React from 'react'

const Context = React.createContext({
    releases: [],    
})

export default Context