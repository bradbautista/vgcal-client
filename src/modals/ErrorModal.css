
.error-heading {
    font-family: 'Roboto Slab', serif;
    font-size: 35px;
    line-height: 35px;
    font-weight: 300;
    padding: 30px;
}

@media screen and (min-width: 700px) {
    
    .error-heading {
        font-size: 42px;
        padding: 30px;
        line-height: 45px;
        letter-spacing: -0.5px;
        font-weight: 200;
    }

}


@media screen and (min-width: 1200px) {
    
    .error-heading {
        font-size: 42px;
        padding: 30px;
        line-height: 45px;
        letter-spacing: -0.5px;
        font-weight: 600;
    }

}