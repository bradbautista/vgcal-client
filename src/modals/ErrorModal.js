import React, { Component } from 'react'
import Context from './../Context';
import './ErrorModal.css';
import Modal from 'react-responsive-modal';
import error from '../images/error.jpg';

  export default class ErrorModal extends Component {
  
    constructor(props) {

      super(props);
      this.state = {
      };
  
    }

    static contextType = Context; 
 
    render() {

        const styles = { 
            overlay: { background: 'rgba(0, 0, 0, 0.35)' },
            modal: { padding: 0, borderRadius: '4px', width: '80%', backgroundColor: 'white' },
            closeButton: { cursor: 'pointer' },
            closeIcon: { fill: 'white', filter: 'drop-shadow( 3px 3px 2px rgba(0, 0, 0, .7))' }
        }
      

        return(

            <Modal 
              open={ this.context.error } 
              onClose={ this.context.onCloseModal }
              styles={styles}
            >
              <img 
                  alt='error' 
                  className='error' 
                  src={error}
              />
                  
              <h2 className='error-heading'>
                  Bad link? vgCal can't find that game, or it doesn't exist. Reload to try again, or give up.
              </h2>
              
            </Modal>
        )
    }
}