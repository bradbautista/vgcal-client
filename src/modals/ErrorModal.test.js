import React from 'react';
import ReactDOM from 'react-dom';
import ErrorModal from './ErrorModal';
import Context from '../Context';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

// Need to use Enzyme for modal snapshot tests b/c of
// react-test-renderer / js-dom issues

const someFunc = () => { console.log('Do stuff') };

it('Renders the error modal without crashing', () => {
    const div = document.createElement('div');
    const someFunc = () => { console.log('Do stuff') };

    ReactDOM.render(<Context.Provider value={ 
        { 
            error: true, 
            onClose: () => someFunc()
        }
    }><ErrorModal /></Context.Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("renders the error modal correctly", () => {

    const subject = mount(
        <BrowserRouter>
            <Context.Provider value={ 
                { 
                    error: true, 
                    onClose: () => someFunc()
                }
            }>
                <ErrorModal />
            </Context.Provider>
        </BrowserRouter>

    );
    expect(EnzymeToJson(subject)).toMatchSnapshot();
});