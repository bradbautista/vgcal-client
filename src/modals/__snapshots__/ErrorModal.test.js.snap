// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the error modal correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ErrorModal>
      <Modal
        animationDuration={500}
        blockScroll={true}
        center={false}
        classNames={Object {}}
        classes={
          Object {
            "closeButton": "styles_closeButton__20ID4",
            "closeIcon": "styles_closeIcon__1QwbI",
            "modal": "styles_modal__gNwvD",
            "modalCenter": "styles_modalCenter__L9F2w",
            "overlay": "styles_overlay__CLSq-",
            "transitionEnter": "styles_transitionEnter__3j_-a",
            "transitionEnterActive": "styles_transitionEnterActive___eQs7",
            "transitionExit": "styles_transitionExit__1KmEf",
            "transitionExitActive": "styles_transitionExitActive__1nQXw",
          }
        }
        closeIconSize={28}
        closeIconSvgPath={
          <path
            d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
          />
        }
        closeOnEsc={true}
        closeOnOverlayClick={true}
        focusTrapOptions={Object {}}
        focusTrapped={true}
        open={true}
        role="dialog"
        showCloseIcon={true}
        styles={
          Object {
            "closeButton": Object {
              "cursor": "pointer",
            },
            "closeIcon": Object {
              "fill": "white",
              "filter": "drop-shadow( 3px 3px 2px rgba(0, 0, 0, .7))",
            },
            "modal": Object {
              "backgroundColor": "white",
              "borderRadius": "4px",
              "padding": 0,
              "width": "80%",
            },
            "overlay": Object {
              "background": "rgba(0, 0, 0, 0.35)",
            },
          }
        }
      >
        <Portal
          containerInfo={
            <div>
              <div
                class="styles_overlay__CLSq- styles_transitionEnter__3j_-a styles_transitionEnterActive___eQs7"
                style="background: rgba(0, 0, 0, 0.35);"
              >
                <div
                  aria-modal="true"
                  class="styles_modal__gNwvD"
                  role="dialog"
                  style="padding: 0px; border-radius: 4px; width: 80%; background-color: white;"
                >
                  <div>
                    <img
                      alt="error"
                      class="error"
                      src="error.jpg"
                    />
                    <h2
                      class="error-heading"
                    >
                      Bad link? vgCal can't find that game, or it doesn't exist. Reload to try again, or try another game.
                    </h2>
                    <button
                      class="styles_closeButton__20ID4"
                      style="cursor: pointer;"
                    >
                      <svg
                        class="styles_closeIcon__1QwbI"
                        height="28"
                        style="fill: white; filter: drop-shadow( 3px 3px 2px rgba(0, 0, 0, .7));"
                        viewBox="0 0 36 36"
                        width="28"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          }
        >
          <CSSTransition
            appear={true}
            classNames={
              Object {
                "appear": "styles_transitionEnter__3j_-a",
                "appearActive": "styles_transitionEnterActive___eQs7",
                "enter": "styles_transitionEnter__3j_-a",
                "enterActive": "styles_transitionEnterActive___eQs7",
                "exit": "styles_transitionExit__1KmEf",
                "exitActive": "styles_transitionExitActive__1nQXw",
              }
            }
            in={true}
            onEntered={[Function]}
            onExited={[Function]}
            timeout={500}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={500}
              unmountOnExit={false}
            >
              <div
                className="styles_overlay__CLSq-"
                onClick={[Function]}
                style={
                  Object {
                    "background": "rgba(0, 0, 0, 0.35)",
                  }
                }
              >
                <div
                  aria-modal="true"
                  className="styles_modal__gNwvD"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": "4px",
                      "padding": 0,
                      "width": "80%",
                    }
                  }
                >
                  <FocusTrap
                    _createFocusTrap={[Function]}
                    active={true}
                    focusTrapOptions={
                      Object {
                        "clickOutsideDeactivates": true,
                      }
                    }
                    paused={false}
                    tag="div"
                  >
                    <div>
                      <img
                        alt="error"
                        className="error"
                        src="error.jpg"
                      />
                      <h2
                        className="error-heading"
                      >
                        Bad link? vgCal can't find that game, or it doesn't exist. Reload to try again, or try another game.
                      </h2>
                      <CloseIcon
                        classNames={Object {}}
                        classes={
                          Object {
                            "closeButton": "styles_closeButton__20ID4",
                            "closeIcon": "styles_closeIcon__1QwbI",
                            "modal": "styles_modal__gNwvD",
                            "modalCenter": "styles_modalCenter__L9F2w",
                            "overlay": "styles_overlay__CLSq-",
                            "transitionEnter": "styles_transitionEnter__3j_-a",
                            "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                            "transitionExit": "styles_transitionExit__1KmEf",
                            "transitionExitActive": "styles_transitionExitActive__1nQXw",
                          }
                        }
                        closeIconSize={28}
                        closeIconSvgPath={
                          <path
                            d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                          />
                        }
                        id={null}
                        onClickCloseIcon={[Function]}
                        styles={
                          Object {
                            "closeButton": Object {
                              "cursor": "pointer",
                            },
                            "closeIcon": Object {
                              "fill": "white",
                              "filter": "drop-shadow( 3px 3px 2px rgba(0, 0, 0, .7))",
                            },
                            "modal": Object {
                              "backgroundColor": "white",
                              "borderRadius": "4px",
                              "padding": 0,
                              "width": "80%",
                            },
                            "overlay": Object {
                              "background": "rgba(0, 0, 0, 0.35)",
                            },
                          }
                        }
                      >
                        <button
                          className="styles_closeButton__20ID4"
                          id={null}
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                            }
                          }
                        >
                          <svg
                            className="styles_closeIcon__1QwbI"
                            height={28}
                            style={
                              Object {
                                "fill": "white",
                                "filter": "drop-shadow( 3px 3px 2px rgba(0, 0, 0, .7))",
                              }
                            }
                            viewBox="0 0 36 36"
                            width={28}
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                            />
                          </svg>
                        </button>
                      </CloseIcon>
                    </div>
                  </FocusTrap>
                </div>
              </div>
            </Transition>
          </CSSTransition>
        </Portal>
      </Modal>
    </ErrorModal>
  </Router>
</BrowserRouter>
`;
